lua/nvim-tree/explorer/watch.lua

```diff
@@ -35,25 +60,17 @@ function M.create_watcher(absolute_path)
     absolute_path = absolute_path,
     interval = M.interval,
     on_event = function(path)
-      local n = utils.get_node_from_path(absolute_path)
-      if not n then
-        return
-      end
-      log.line("watcher", "node event '%s'", path)
-
-      local node = utils.get_parent_of_group(n)
-      local project_root, project = reload_and_get_git_project(path)
-      require("nvim-tree.explorer.reload").reload(node, project)
-      update_parent_statuses(node, project, project_root)
-
-      require("nvim-tree.renderer").draw()
+      utils.debounce("explorer:watcher:" .. path, M.debounce, function()
+        refresh_path(path)
+      end)
     end,
   }
 end
 
 function M.setup(opts)
   M.enabled = opts.filesystem_watchers.enable
   M.interval = opts.filesystem_watchers.interval
+  M.debounce = opts.filesystem_watchers.debounce
```

Comment:
i'd call this debounce_timeout or something like that :)

Reply:
`debounce_delay`